octopus: rgw/notification: trigger notifications on changes from any user

yuvalif opened the pull request:
backport tracker: https://tracker.ceph.com/issues/48461

backport of https://github.com/ceph/ceph/pull/38136

as well as fixes to manual notification tests, in order to change the main change

smithfarm commented:
@yuvalif Typically the reviewers are pretty relaxed about this stuff, so this PR might be deemed "fine as it is".

Still, there seems to be some minor confusion surrounding cherry-picking which I'll attempt to clear up.

First, I would suggest to quickly skim over the cherry-pick rules here: https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules

In accordance with that, here is how I personally would have done this backport:

1. cherry-pick the master commit - `git cherry-pick -x 658993efc16498c2106108fe407f6d44040f349d`
2. refrain from changing the master commit message (in this case you prefixed the commit title with `octopus:`, but that is not called for)
3. squash the additional testing commit into the cherry-pick commit and describe it in the `Conflicts` section.

That way, the number of commits in the backport PR will be the same as in the master PR.

Also, it would help the Stable Releases team if you would paste the backport tracker issue URL in the backport PR description (I went ahead and did that for you).

Hope this helps!

yuvalif commented:
> @yuvalif Typically the reviewers are pretty relaxed about this stuff, so this PR might be deemed "fine as it is".
> 
> Still, there seems to be some minor confusion surrounding cherry-picking which I'll attempt to clear up.
> 
> First, I would suggest to quickly skim over the cherry-pick rules here: https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules
> 
> In accordance with that, here is how I personally would have done this backport:
> 
> 1. cherry-pick the master commit - `git cherry-pick -x 658993efc16498c2106108fe407f6d44040f349d`
> 2. refrain from changing the master commit message (in this case you prefixed the commit title with `octopus:`, but that is not called for)

fixed
> 3. squash the additional testing commit into the cherry-pick commit and describe it in the `Conflicts` section.
> 
fixed
> That way, the number of commits in the backport PR will be the same as in the master PR.
> 
> Also, it would help the Stable Releases team if you would paste the backport tracker issue URL in the backport PR description (I went ahead and did that for you).
> 
> Hope this helps!

yes, it does, thanks!
